### PR TITLE
-Wall and -Wextra compile options

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
Probably best to review the commits separately

### CODE REVIEW

File: src/sensors/DepthCamera.cpp [hunk 1]
```diff
@@ -279,7 +278,7 @@ void DepthCamera::drawTriangle2D(const geo::Vec3T<T>& p1, const geo::Vec3T<T>& p
 void DepthCamera::drawTrianglePart(int y_start, int y_end,
                                    float x_start, float x_start_delta, float x_end, float x_end_delta,
                                    float d_start, float d_start_delta, float d_end, float d_end_delta,
-                                   const RenderOptions& opt, RenderResult& res, uint i_triangle) const {
+                                   const RenderOptions& /*opt*/, RenderResult& res, uint i_triangle) const {
```
Comment (ar13pit) [inline src/sensors/DepthCamera.cpp:281]: Why not remove the arg completely?

File: src/Shape.cpp [hunk 2]
```diff
@@ -307,7 +305,7 @@ bool Shape::contains(const Vector3& p) const {
                         i1 = it->i2_;
                         i2 = it->i3_;
                     }
-                    else if (std::abs(s3) < 1e-16) {
+                    else { // if (std::abs(s3) < 1e-16)
```
Comment (ar13pit) [inline src/Shape.cpp:308]: Remove commented code through out

**Review by ar13pit — CHANGES_REQUESTED** [review]
Also squash commits :)

File: src/Shape.cpp (hunk 2 above)
Comment (MatthijsBurgh) [reply src/Shape.cpp:308]: I want this to be clear what the actual condition is when it gets to this point

File: src/sensors/DepthCamera.cpp (hunk 1 above)
Comment (MatthijsBurgh) [reply src/sensors/DepthCamera.cpp:281]: So we keep a similar API through all functions

**Comment by MatthijsBurgh** [comment]
> Also squash commits :)

Which ones, I don't agree to squash everything.

File: src/sensors/DepthCamera.cpp (hunk 1 above)
Comment (MatthijsBurgh) [reply src/sensors/DepthCamera.cpp:281]: So when this code changes in the future and it does get used, the order of arguments is not different.

**Comment by ar13pit** [comment]
> > Also squash commits :)
> 
> Which ones, I don't agree to squash everything.

All the unused parameters ones into one. CMake changes into one. Copy constructor and default constructor ones into one.

**Comment by MatthijsBurgh** [comment]
> > > Also squash commits :)
> > 
> > 
> > Which ones, I don't agree to squash everything.
> 
> All the unused parameters ones into one. CMake changes into one. Copy constructor and default constructor ones into one.

Done, except for the constructor changes. As these have a different rule as cause.

**Review by ar13pit — APPROVED** [review]

**Pull request merged**